src/XRP/reliable-submission-xrp-client.ts

```diff
@@ -48,12 +49,72 @@ export default class ReliableSubmissionXrpClient implements XrpClientDecorator {
     sendXrpDetails: SendXrpDetails,
   ): Promise<string> {
     const { sender } = sendXrpDetails
-    const ledgerCloseTimeMs = 4 * 1000
 
-    // Submit a transaction hash and wait for a ledger to close.
     const transactionHash = await this.decoratedClient.sendWithDetails(
       sendXrpDetails,
     )
+    await this.awaitFinalTransactionResult(transactionHash, sender)
+
+    return transactionHash
+  }
+
+  public async getLatestValidatedLedgerSequence(
+    address: string,
+  ): Promise<number> {
+    return this.decoratedClient.getLatestValidatedLedgerSequence(address)
+  }
+
+  public async getRawTransactionStatus(
+    transactionHash: string,
+  ): Promise<RawTransactionStatus> {
+    return this.decoratedClient.getRawTransactionStatus(transactionHash)
+  }
+
+  public async accountExists(address: string): Promise<boolean> {
+    return this.decoratedClient.accountExists(address)
+  }
+
+  public async paymentHistory(address: string): Promise<Array<XrpTransaction>> {
+    return this.decoratedClient.paymentHistory(address)
+  }
+
+  public async getPayment(
+    transactionHash: string,
+  ): Promise<XrpTransaction | undefined> {
+    return this.decoratedClient.getPayment(transactionHash)
+  }
+
+  public async enableDepositAuth(wallet: Wallet): Promise<TransactionResult> {
+    const result = await this.decoratedClient.enableDepositAuth(wallet)
+    const transactionHash = result.hash
+
+    const rawTransactionStatus = await this.awaitFinalTransactionResult(
+      transactionHash,
+      wallet,
+    )
+
+    // Return pending if the transaction is not validated.
```

Comment:
Sorry if I missed this in our other libs but should we refactor this logic to a helper (`determineFinalResult(rawTransactionStatus)`)